app/src/pages/playground/schemas.ts

```diff
@@ -182,6 +188,67 @@ export const modelConfigWithInvocationParametersSchema = z.object({
   }),
 });
 
+/**
+ *  The zod schema for llm.tools.{i}.tool.json_schema attribute
+ *  This will be a json string but should be parsed as an object
+ */
+export const toolJSONSchemaSchema = z
+  .string()
+  .transform((s, ctx) => {
+    const { json } = safelyParseJSON(s);
+
+    if (json == null || !isObject(json)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "The tool JSON schema must be a valid JSON object",
+      });
+      return z.NEVER;
+    }
+    return json;
+  })
+  // TODO(parker / apowell) - adjust this transformation with anthropic tool support https://github.com/Arize-ai/phoenix/issues/5100
+  .transform((o, ctx) => {
+    const { data, success } = openAIToolSchema.safeParse(o);
+
+    if (!success) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: "The tool JSON schema must be a valid OpenAI tool schema",
+      });
+      return z.NEVER;
+    }
+    return data;
+  });
+
+/**
+ * The zod schema for llm.tools
+ * @see {@link  https://github.com/Arize-ai/openinference/blob/main/spec/semantic_conventions.md|Semantic Conventions}
+ * Note there are other attributes that can be on llm.tools.{i}.tool, namely description, name, and parameters
+ * however, these are encompassed by the json schema in some cases and calls to api's using tools in a non json format
+ * is not supported in the playground yet
+ */
+export const llmToolSchema = z
+  .object({
+    [SemanticAttributePrefixes.llm]: z
+      .object({
+        [LLMAttributePostfixes.tools]: z
+          .array(
+            z
+              .object({
+                [SemanticAttributePrefixes.tool]: z.object({
+                  [ToolAttributePostfixes.json_schema]: toolJSONSchemaSchema,
+                }),
+              })
+              .optional()
```

Comment:
made everything optional here and filtered before passing to the playground so we can capture some tools even if some are malformed